Sources/Testing/SourceAttribution/Backtrace.swift

```diff
@@ -183,9 +188,15 @@ extension Backtrace {
 
   /// The implementation of ``Backtrace/startCachingForThrownErrors()``, run
   /// only once.
-  private static let _startCachingForThrownErrors: Void = {
+  ///
+  /// This value is named oddly so that it shows up clearly in symbolicated
+  /// backtraces.
+  private static let __SWIFT_TESTING_IS_CAPTURING_A_BACKTRACE_FOR_A_THROWN_ERROR__: Void = {
```

Comment:
Do you need to customize anything further to get this to be the "raw" linker symbol name for this API? Wondering if the compiler might mangle this name at all.

I'm curious what an example backtrace with this PR in effect looks like, in any case, if you have one handy to share.

Reply:
This is sufficient for the string we care about to show up in a stack trace—I didn't deem further customization e.g. with `@_silgen_name` to be worth the fuss.

Example backtrace with this change in place in Xcode 16 Beta:

```
#0	0x00000001077f9ea4 in static Backtrace._willThrow(_:from:) at /Volumes/Dev/Source/swift-testing/Sources/Testing/SourceAttribution/Backtrace.swift:174
#1	0x00000001077fa364 in closure #1 in closure #1 in closure #1 in variable initialization expression of static Backtrace.__SWIFT_TESTING_IS_CAPTURING_A_BACKTRACE_FOR_A_THROWN_ERROR__ at /Volumes/Dev/Source/swift-testing/Sources/Testing/SourceAttribution/Backtrace.swift:198
#2	0x00000001077fa384 in @objc closure #1 in closure #1 in closure #1 in variable initialization expression of static Backtrace.__SWIFT_TESTING_IS_CAPTURING_A_BACKTRACE_FOR_A_THROWN_ERROR__ ()
#3	0x000000019b85bccc in swift_willThrow ()
#4	0x00000001068ef944 in closure #9 in closure #2 in IssueTests.testErrorCheckingWithExpect_Mismatching() at /var/folders/31/7fsv9vpn7r793cpf_32mbj_c0000gn/T/swift-generated-sources/@__swiftmacro_12TestingTests0021IssueTestsswift_IfFDefMX572_6_33_777AD2679C7C3B7F621B0C27AB573F75Ll6expectfMf2_.swift:2
#5	0x000000010777ab58 in __checkClosureCall<()>(performing:throws:mismatchExplanation:expression:comments:isRequired:sourceLocation:) at /Volumes/Dev/Source/swift-testing/Sources/Testing/Expectations/ExpectationChecking+Macro.swift:1050
#6	0x0000000107779d38 in __checkClosureCall<TestingTests.MyError, ()>(throws:performing:expression:comments:isRequired:sourceLocation:) at /Volumes/Dev/Source/swift-testing/Sources/Testing/Expectations/ExpectationChecking+Macro.swift:847
#7	0x00000001068eed2c in closure #2 in IssueTests.testErrorCheckingWithExpect_Mismatching() at /var/folders/31/7fsv9vpn7r793cpf_32mbj_c0000gn/T/swift-generated-sources/@__swiftmacro_12TestingTests0021IssueTestsswift_IfFDefMX572_6_33_777AD2679C7C3B7F621B0C27AB573F75Ll6expectfMf2_.swift:1
...
```

This is _without_ optimizations in a debug build; the stack is shorter when optimizations are enabled. (The `@objc` thunk in particular is an artifact.)